controllers/submariner/loadbalancer_resources.go

```diff
@@ -43,18 +48,74 @@ const (
 func (r *Reconciler) reconcileLoadBalancer(
 	ctx context.Context, instance *v1alpha1.Submariner, reqLogger logr.Logger,
 ) (*corev1.Service, error) {
-	return apply.Service(ctx, instance, newLoadBalancerService(instance), reqLogger, r.config.ScopedClient, r.config.Scheme)
+	var err error
+
+	platformTypeOCP, err := r.getOCPPlatformType(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	svc := newLoadBalancerService(instance, platformTypeOCP)
+	svc, err = apply.Service(ctx, instance, svc, reqLogger, r.config.ScopedClient, r.config.Scheme)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// For IBM cloud also needs to annotate the allocated health check node port
+	if platformTypeOCP == string(configv1.IBMCloudPlatformType) {
+		healthPortStr := fmt.Sprintf("%d", svc.Spec.HealthCheckNodePort)
+		svc.ObjectMeta.Annotations = map[string]string{
+			"service.kubernetes.io/ibm-load-balancer-cloud-provider-vpc-health-check-port": healthPortStr,
+		}
+		svc, err = apply.Service(ctx, instance, svc, reqLogger, r.config.ScopedClient, r.config.Scheme)
+	}
+
+	return svc, err
 }
 
-func newLoadBalancerService(instance *v1alpha1.Submariner) *corev1.Service {
+func (r *Reconciler) getOCPPlatformType(ctx context.Context) (string, error) {
+	clusterInfra := &configv1.Infrastructure{}
+	err := r.config.GeneralClient.Get(ctx, types.NamespacedName{Name: "cluster"}, clusterInfra)
+
+	if apierrors.IsNotFound(err) {
+		return "", nil
+	}
+
+	if err != nil {
+		return "", errors.Wrap(err, "error retrieving cluster Infrastructure resource")
+	}
+
+	if clusterInfra.Status.PlatformStatus == nil {
+		return "", nil
+	}
```

Comment:
For backwards compatibility, perhaps we should return the deprecated `Platform` field if the newer field isn't set:

```suggestion
	if clusterInfra.Status.PlatformStatus == nil {
		return string(clusterInfra.Platform), nil
	}
```

Reply:
> For backwards compatibility, perhaps we should return the deprecated `Platform` field if the newer field isn't set:

Could be (it should be `return string(clusterInfra.Status.Platform), nil` ) , but then we'll get below linter error, so not sure about this change.

```
controllers/submariner/loadbalancer_resources.go:90:17: SA1019: clusterInfra.Status.Platform is deprecated: Use platformStatus.type instead. (staticcheck)
                return string(clusterInfra.Status.Platform), nil
```

Reply:
Since we’d be using a deprecated field on purpose, it would be find to disable the check (`//nolint:staticcheck`).